ibis/backends/dask/client.py

```diff
@@ -156,18 +157,7 @@ class DaskDatabase(Database):
     pass
 
 
-class DaskClient(Client):
-    def __init__(self, backend, dictionary: Dict[str, dd.DataFrame]):
-        self.backend = backend
-        self.database_class = backend.database_class
-        self.table_class = backend.table_class
-        self.dictionary = dictionary
-
-    def table(self, name: str, schema: sch.Schema = None) -> DaskTable:
-        df = self.dictionary[name]
-        schema = sch.infer(df, schema=schema)
-        return self.table_class(name, schema, self).to_expr()
-
+class DaskClient(PandasClient):
```

Comment:
I'm not sure it makes sense conceptually to answer "yes" to the question "Is a dask client a pandas client?"

I don't think deduplicating code is a strong enough rationale for coupling these two components.

Now, `DaskClient`s would inherit any change to `PandasClient` APIs that are now duplicated.

I could probably get behind a more specific parent class of both `PandasClient` and `DaskClient`, like `DataFrameClient`, where the duplication was reduced but pandas and dask aren't coupled.

Reply:
Agree, that's how it is implemented for `Backend`. The name is `BasePandasBackend`, since it's not a generic dataframe backend, but a pandas-API base backend.

Whenever we move the methods to `Backend`, they'll have the structure you mention. I'm ok not merging this, and consolidating the methods when we move to `Backend`, but I thought having this early should simplify the other PR.

Reply:
I see. That makes sense then. Can you add a `TODO` with a some of what you've written here? Maybe something like:

```
TODO: these are coupled here to remove duplication, and are uncoupled in the Backend refactor
```